### ISSUE TITLE
Topic.publish inside timer.cron blows up

### ISSUE THREAD

**Issue by mikhailshilkov** [pulumi/pulumi-cloud#612]
I have the following code:

```
const topic = new cloud.Topic("fire");
cloud.timer.cron("schedulefire", "0 0 * * * *", async () => {
    topic.publish(new FireMessage("blabla", 10));
});
```

it blows up during `pulumi update` with a whole loads of text aboud serializing things:

```
    error: Error serializing '(context, data) => { // Redirect con ...': subscription.js(64,11)

    '(context, data) => { // Redirect con ...': subscription.js(64,11): captured
      'callback', a function defined at
        '(context, data) => { handler().then( ...': timer.js(117,214): which captured
          'handler', a function defined at
            '() => __awaiter(this, void 0, void 0 ...': index.ts(65,48): which captured
              variable 'topic' which indirectly referenced
                '(val) => __awaiter(this, void 0, voi ...': topic.js(51,23): which captured
                  module './node_modules\\azure-sb\\lib\\servicebus.js' which indirectly referenced
                    function '<anonymous>': servicebus.js(38,43): which captured
                      module './node_modules\\azure-sb\\lib\\servicebusservice.js' which indirectly referenced
                        function '<anonymous>': serviceclient.js(86,50): which captured
                          module './node_modules\\azure-common\\node_modules\\underscore\\underscore.js' which indirectly referenced
                            function '<anonymous>': underscore.js(811,20): which captured
                              module './node_modules\\azure-common\\node_modules\\underscore\\underscore.js' which indirectly referenced
                                function 'isArray': which could not be serialized because
                                  it was a native code function.

    Function code:
      function isArray() { [native code] }

    Capturing modules can sometimes cause problems.
    Consider using import('./node_modules\\azure-sb\\lib\\servicebus.js') or require('./node_modules\\azure-sb\\lib\\servicebus.js') inside '(val) => __awaiter(this, void 0, voi ...': topic.js(51,23)

    error: Error serializing '(context, data) => { // Redirect con ...': subscription.js(64,11)

    '(context, data) => { // Redirect con ...': subscription.js(64,11): captured
      'callback', a function defined at
        '(context, data) => { handler().then( ...': timer.js(117,214): which captured
          'handler', a function defined at
            '() => __awaiter(this, void 0, void 0 ...': index.ts(65,48): which captured
              variable 'topic' which indirectly referenced
                '(val) => __awaiter(this, void 0, voi ...': topic.js(51,23): which captured
                  module './node_modules\\azure-sb\\lib\\servicebus.js' which indirectly referenced
                    function '<anonymous>': servicebus.js(38,43): which captured
                      module './node_modules\\azure-sb\\lib\\servicebusservice.js' which indirectly referenced
                        function '<anonymous>': serviceclient.js(86,50): which captured
                          module './node_modules\\azure-common\\node_modules\\underscore\\underscore.js' which indirectly referenced
                            function '<anonymous>': underscore.js(811,20): which captured
                              module './node_modules\\azure-common\\node_modules\\underscore\\underscore.js' which indirectly referenced
                                function 'isArray': which could not be serialized because
                                  it was a native code function.

    Function code:
      function isArray() { [native code] }

    Capturing modules can sometimes cause problems.
    Consider using import('./node_modules\\azure-sb\\lib\\servicebus.js') or require('./node_modules\\azure-sb\\lib\\servicebus.js') inside '(val) => __awaiter(this, void 0, voi ...': topic.js(51,23)

    error: an unhandled error occurred: Program exited with non-zero exit code: 1
```

Any way I should do it differently?

Pulumi v `0.15.2`, all dependencies on `latest`, cloud is Azure.

**Comment by joeduffy** [comment]
This is going to require @CyrusNajmabadi's wisdom, I think.

This appears to be triggered by the way Underscore.js defines the `isArray` function, which the Azure SDK is using. E.g.,

https://github.com/jashkenas/underscore/blob/d5fe0fd4060f13b40608cb9d92eda6d857e8752c/underscore.js#L1315

which is defined as

https://github.com/jashkenas/underscore/blob/d5fe0fd4060f13b40608cb9d92eda6d857e8752c/underscore.js#L34

which is a native function, and cannot be serialized.

There are two things I don't understand

1. Where is the Azure SDK being referenced? Not in your code, right? So, I assume this is in the definition of the cloud SDK somewhere.

2. Why are we trying to serialize the Azure SDK module at all? I thought we would have turned this into a true module import on the other side of the function?

Note that this part is suspicious to me

```
 'handler', a function defined at
            '() => __awaiter(this, void 0, void 0 ...': index.ts(65,48): which captured
              variable 'topic' which indirectly referenced
```

I thought we had previously done something funny around `__awaiter` functions, to avoid capturing the `this` variable because in that context, it captured the entire module in scope at that time.

TL;DR, I believe your code should just work, and that there's a bug in our code here, but @CyrusNajmabadi will need to verify.

**Comment by CyrusNajmabadi** [comment]
Unfortunately this is likely something we need to fix on our end. I have an idea about a potential workaround, but I'm not in front ofa computer to test it out. Basically, we need to mark a method on topic as not to be serialized

**Comment by CyrusNajmabadi** [comment]
I'm looking at this now.

**Comment by CyrusNajmabadi** [comment]
hrmm... couldn't repro this with a simple example.  @mikhailshilkov Where does "FireMessage" come from?  can you share your full repro file?   I'll keep looking to see if i can figure out what's going on here.

**Comment by CyrusNajmabadi** [comment]
Ok.  I think i know what the issue is.  The key line is this:

```
                '(val) => __awaiter(this, void 0, voi ...': topic.js(51,23): which captured
                  module './node_modules\\azure-sb\\lib\\servicebus.js' which indirectly referenced
```

Basically, this shows that we're trying to serialize the 'azure-sb' module 'by valaue' instead of just inserting it as `require("azure-sb")`.  I'm fairly confident hte reason we're doing that is because we're screwing up our paths between `/`-unix paths and `\`-windows paths.

Specifically, because the path starts with `./node_modules\\` we don't think it's an actual node_module path due to this specific check: https://github.com/pulumi/pulumi/blob/5ed33c691509571c03cbf44f22c6f05391f238ba/sdk/nodejs/runtime/closure/createClosure.ts#L1090

I can definitely work on fixing this.  But i'd love some help here @mikhailshilkov since you're on Windows :)  Let me see if i can whip up a fix, and we'll try it out with you asap.
